rad/src/operators/map.rs

```diff
@@ -26,33 +25,51 @@ pub fn get(input: &RadonMap, args: &[Value]) -> Result<RadonTypes, RadError> {
         .map(Clone::clone)
         .ok_or_else(|| not_found(key))
 }
-pub fn get_array(input: &RadonMap, args: &[Value]) -> Result<RadonArray, RadError> {
-    let item = get(input, args)?;
-    item.try_into()
-}
-pub fn get_boolean(input: &RadonMap, args: &[Value]) -> Result<RadonBoolean, RadError> {
-    let item = get(input, args)?;
-    item.try_into()
-}
-pub fn get_bytes(input: &RadonMap, args: &[Value]) -> Result<RadonBytes, RadError> {
-    let item = get(input, args)?;
-    item.try_into()
-}
-pub fn get_integer(input: &RadonMap, args: &[Value]) -> Result<RadonInteger, RadError> {
-    let item = get(input, args)?;
-    item.try_into()
-}
-pub fn get_float(input: &RadonMap, args: &[Value]) -> Result<RadonFloat, RadError> {
-    let item = get(input, args)?;
-    item.try_into()
+
+/// Try to get any kind of `RadonType` from an entry in the input `RadonArray`, as specified
+/// by the first argument, which is used as the search key.
+pub fn get<O: RadonType<T>, T>(input: &RadonMap, args: &[Value]) -> Result<O, RadError>
+where
+    T: std::fmt::Debug,
+{
+    let item = inner_get(input, args)?;
+    let original_type = item.radon_type_name();
+
+    item.try_into().map_err(|_| RadError::Decode {
+        from: original_type,
+        to: O::radon_type_name(),
+    })
 }
-pub fn get_map(input: &RadonMap, args: &[Value]) -> Result<RadonMap, RadError> {
-    let item = get(input, args)?;
-    item.try_into()
+
+/// Try to get a `RadonFloat` or  `RadonInteger` from an entry in the input `RadonMap`, as specified
+/// by the first argument, which is used as the search key. Internally does some pre-processing
+/// to normalize decimal and thousands separators.
+pub fn get_number<O>(input: &RadonMap, args: &[Value]) -> Result<O, RadError>
+where
+    O: TryFrom<RadonTypes, Error = RadError>,
+{
+    let original_type = inner_get(input, &args[..1])?.radon_type_name();
+
+    get_numeric_string(input, args)
+        .map(RadonTypes::from)
+        .and_then(O::try_from)
+        .map_err(|err| err.replace_decode_from(original_type))
 }
-pub fn get_string(input: &RadonMap, args: &[Value]) -> Result<RadonString, RadError> {
-    let item = get(input, args)?;
-    item.try_into()
+
+/// Try to get a `RadonTypes` from an entry in the input `RadonMap`, as specified by the first
+/// argument, which is used as the search key.
+///
+/// This simply assumes that the element in that position is a number (i.e., `RadonFloat` or
+/// `RadonInteger`). If it is not, it will fail with a `RadError` because of `replace_separators`.
+fn get_numeric_string(input: &RadonMap, args: &[Value]) -> Result<RadonString, RadError> {
+    let item = get::<RadonString, _>(input, &args[..1])?.value();
+    let (decimal_separator, thousands_separator) = string::read_separators_from_args(&args[1..]);
+
+    Ok(RadonString::from(string::replace_separators(
+        item,
+        decimal_separator,
+        thousands_separator,
+    )))
```

Comment:
These names are swapped, the tests pass because they are swapped twice.

```suggestion
    let (thousands_separator, decimal_separator) = string::read_separators_from_args(&args[1..]);

    Ok(RadonString::from(string::replace_separators(
        item,
        thousands_separator,
        decimal_separator,
    )))
```

Reply:
Lol. I blame it on CLion's automatic name refactors.